### ISSUE TITLE
Regression: 6.10.0 Search with _count=MAX_INT overflows

### ISSUE THREAD

**Issue by jhahkala** [hapifhir/hapi-fhir#5497]
NOTE: Before filing a ticket, please see the following URL:
https://github.com/hapifhir/hapi-fhir/wiki/Getting-Help

**Describe the bug**
In HAPI JPA server 6.8.5 and before search with parameter _count=2147483647 (MAX_INT) worked. In version 6.10.0 one gets internal server error and "diagnostics": "ERROR: LIMIT must not be negative" which comes from postgres when the limit is negative. In version 6.10.0 _count=2147483646 (MAX_INT-1) works. So, my guess is that somewhere in the code there is a +1 added to the  count/limit. I tried to track it down, but couldn't.

**To Reproduce**
1. Go to http://{host}/fhir/Patient/_search?_count=2147483647 on a server that uses 6.10.0 version
2. See error

**Expected behavior**
List of Patients, like you get with http://{host}/fhir/Patient/_search?_count=2147483646

**Environment (please complete the following information):**
 - HAPI FHIR Version 6.10.0
 - OS: eclipse-temurin:17-jre-alpine

**Comment by jamesagnew** [comment]
I'm not seeing this behaviour on our public server: http://hapi.fhir.org/baseR4/Patient?_count=2147483647

**Comment by jhahkala** [comment]
True, but the test server does use 6.11.2-SNAPSHOT. I tried earlier the public server but then everything was failing with the timer issue  #5485, #5479, so, I couldn't verify. Maybe it's specific to our setup.

**Comment by jhahkala** [comment]
Hi,

I traced down the change. In the commit https://github.com/hapifhir/hapi-fhir/commit/fdfda37f7371594f568a6670c146a96562670b86# in SearchTask line 607 there is +1 added to the myMaxResultsToFetch, which then overflows. The commit is included in 6.10.0. I don't know why the problem doesn't appear in the snapshot in the public test server.

**Comment by jhahkala** [comment]
I don't know if the search even has to work with the _count=2147483647. We have changed our limit, so it won't bother us anymore :smile: Feel free to close the bug if the max integer value doesn't have to be supported.
